### PR TITLE
Mailbox 268 Store in Cassandra attachments ids related to a message

### PR DESCRIPTION
**Pull request by rouazana**
Based on antoine's PR, so only the 2 last commits are for review here.

### CODE REVIEW

File: mailbox/store/src/test/java/org/apache/james/mailbox/store/mail/model/AbstractMessageMapperTest.java [hunk 1]
```diff
@@ -249,6 +258,33 @@ public void messagesRetrievedUsingFetchTypeBodyShouldHaveBodyDataLoaded() throws
...
+        Iterator<MailboxMessage> retrievedMessageIterator = messageMapper.findInMailbox(benwaInboxMailbox, MessageRange.one(message1.getUid()), fetchType, LIMIT);
+        MessageAssert.assertThat(retrievedMessageIterator.next()).isEqualTo(message1, fetchType);
+        assertThat(retrievedMessageIterator).isEmpty();
+    }
+
+    @Test
+    public void messagesRetrievedUsingFetchTypeFullShouldHaveAttachmentsIdsLoadedWhenOneAttachment() throws MailboxException, IOException{
+        saveMessages();
+        MessageMapper.FetchType fetchType = MessageMapper.FetchType.Full;
+        Iterator<MailboxMessage> retrievedMessageIterator = messageMapper.findInMailbox(attachmentsMailbox, MessageRange.one(message7With1Attachment.getUid()), fetchType, LIMIT);
```
Comment (chibenwa) [inline mailbox/store/src/test/java/org/apache/james/mailbox/store/mail/model/AbstractMessageMapperTest.java]: Is it only true with Fetch type Full ?

For which type of Fetch Type could the attachement Ids not be retrieved ?

Comment (rouazana) [inline mailbox/store/src/test/java/org/apache/james/mailbox/store/mail/model/AbstractMessageMapperTest.java]: Good question. I return it only if FetchType is full. But I have no idea if that's the good thing to do.
What's your mind?

Comment (rouazana) [inline mailbox/store/src/test/java/org/apache/james/mailbox/store/mail/model/AbstractMessageMapperTest.java]: Anyway I should add a test indicating that it is not retrieved if fetchType is not full, and maybe simplify these one to only check attachmentsIds and not the other fields.

Comment (rouazana) [inline mailbox/store/src/test/java/org/apache/james/mailbox/store/mail/model/AbstractMessageMapperTest.java]: Tests reworked.
Please tell me if you think we should return AttachmentsIds for other fetch types.

Comment (chibenwa) [inline mailbox/store/src/test/java/org/apache/james/mailbox/store/mail/model/AbstractMessageMapperTest.java]: For body also IMO

Comment (rouazana) [inline mailbox/store/src/test/java/org/apache/james/mailbox/store/mail/model/AbstractMessageMapperTest.java]: thanks, added

**Comment by aduprat** [comment]
:+1:

**Comment by chibenwa** [comment]
:+1:

File: mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageMapper.java [hunk 2]
```diff
@@ -328,6 +332,19 @@ private PropertyBuilder getPropertyBuilder(Row row) {
...
+    private List<AttachmentId> getAttachmentsIds(Row row, FetchType fetchType) {
+        switch (fetchType) {
+        case Full:
+        case Body:
+            return row.getList(ATTACHMENTS_IDS, String.class)
+                    .stream()
+                    .map(AttachmentId::from)
+                    .collect(org.apache.james.util.streams.Collectors.toImmutableList());
+        default:
+            return ImmutableList.of();
```
Comment (mbaechler) [inline mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageMapper.java]: how do you make a difference between no attachments and attachments not loaded then ?

Comment (rouazana) [inline mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageMapper.java]: I don't think we need to. It is the same principle in buildContent() for example.

**Pull request merged**